iai-callgrind-runner/src/runner/tool/logfile_parser.rs

```diff
@@ -49,17 +51,79 @@ pub struct LogfileSummary {
     pub fields: Vec<(String, String)>,
     pub details: Vec<String>,
     pub error_summary: Option<ErrorSummary>,
+    pub costs: Option<Costs<String>>,
     pub log_path: PathBuf,
 }
-
 #[derive(Debug, PartialEq, Eq, Clone, Copy)]
 enum State {
     Header,
     HeaderSpace,
     Body,
 }
 
-impl LogfileParser {
+pub trait LogfileParserT {
+    fn parse_single(&self, path: PathBuf) -> Result<LogfileSummary>;
+
+    fn parse(&self, output_path: &ToolOutputPath) -> Result<Vec<LogfileSummary>> {
+        let log_path = output_path.to_log_output();
+        debug!("{}: Parsing log file '{}'", output_path.tool.id(), log_path);
+
+        let mut summaries = vec![];
+        let Ok(paths) = log_path.real_paths() else {
+            return Ok(vec![]);
+        };
+        for path in paths {
+            let summary = self.parse_single(path)?;
+            summaries.push(summary);
+        }
+        summaries.sort_by_key(|x| x.pid);
+        Ok(summaries)
+    }
+}
+
+pub fn merge_logfile_summaries(
```

Comment:
The way I understand this the `ToolRunSummary` is supposed to represent the new run possibly being compared against the old run. If the commands don't match I would expect either an error or a single `ToolRunSummary` based only on the new `LogfileSummary` while the old `LogfileSummary` is ignored/

Reply:
You're right and I'm sorry, but for the time being this method has to work for most cases. Like I said you can keep it simple and create a `ToolRunSummary` for each of the diverging `LogfileSummary`. I think that's the best we can do right now. Imagine the situation when the commands don't match but someone sees dhat differences in the output. There's no way to tell why there are differences because there's only one command shown, for example `Command: /first` but in reality we created a diff between `Command: /first` and `Command: /second --some --arg`. This is very misleading and it would be much clearer if the comparison shows (I'm simplifying here)

```
Command: /first
Total: 1000|N/A (****)

Command: /second --some --arg
Total: N/A|500 (****)
````

Reply:
Would it be acceptable to only show:
```
Command: /first
Total: 1000|N/A (****)
```
The way I have things set up isn't designed to have `ToolRunSummary` that only includes old run but not the new run.

Reply:
For `pid` you can fill in `0` for example. The `costs` aren't optional for `new` but can't you just use Costs with an empty IndexMap to create a `CostsSummary`? Are there other fields of `ToolRunSummary` which can't be filled from `old`?

Reply:
If the length of the old list is longer than the new list should I also include the extra old logfile summaries? Would it be better to make the `pid` field of `ToolRunSummary` an `Option` rather that using `0`? How should this be handle for tools other than DHAT?

Reply:
Thanks :) Most of it look good now.